diff --git a/cmake/ide/segger.cmake b/cmake/ide/segger.cmake
index a36abdf7..940d6c35 100644
--- a/cmake/ide/segger.cmake
+++ b/cmake/ide/segger.cmake
@@ -911,6 +911,14 @@ function(generate_ses_project)
             set(is_secondary_core 1)
         endif()
     endif()
+    if (DEFINED PROJECT_TYPE_APP)
+        if (DEFINED PROJECT_APP_TYPE_UPGRADE)
+            set(post_build_command "${PYTHON_EXECUTABLE} ${MCUBOOT_ROOT_DIR}/scripts/imgtool.py sign --header-size ${MCUBOOT_IMAGE_HEADER_SIZE} --align 8 --version ${PROJECT_APP_VERSION} --slot-size ${MCUBOOT_APPLICATION_SIZE} --pad $(OutDir)/demo.bin $(OutDir)/demo_post.bin")
+        else()
+            set(post_build_command "${PYTHON_EXECUTABLE} ${MCUBOOT_ROOT_DIR}/scripts/imgtool.py sign --header-size ${MCUBOOT_IMAGE_HEADER_SIZE} --align 8 --version ${PROJECT_APP_VERSION} --slot-size ${MCUBOOT_APPLICATION_SIZE} $(OutDir)/demo.bin $(OutDir)/demo_post.bin")
+        endif()
+        string(REPLACE "\\" "/" post_build_command ${post_build_command})
+    endif()
 
     set(target_link_symbols "")
     foreach(sym IN ITEMS ${target_link_sym})
